pkg/odo/cli/url/create.go

```diff
@@ -77,14 +79,23 @@ func (o *URLCreateOptions) Validate() (err error) {
 // Run contains the logic for the odo url create command
 func (o *URLCreateOptions) Run() (err error) {
 
-	log.Infof("Adding URL to component: %v", o.Component())
 	urlRoute, err := url.Create(o.Client, o.urlName, o.componentPort, o.Component(), o.Application)
 	if err != nil {
 		return err
 	}
 	urlCreated := url.GetURLString(urlRoute.Spec.Protocol, urlRoute.Spec.URL)
-	log.Successf("URL created for component: %v\n\n"+
-		"%v - %v\n", o.Component(), urlRoute.Name, urlCreated)
+
+	if o.outputFlag == "json" {
+		out, err := json.Marshal(urlRoute)
+		if err != nil {
+			return err
+		}
+		fmt.Println(string(out))
+	} else {
+		log.Infof("Adding URL to component: %v", o.Component())
```

Comment:
This line should be before the url.create() call

Reply:
I removed the log line, it is not much helpful